gdal/swig/include/ogr.i

```diff
@@ -853,6 +853,10 @@ public:
     return layer;
   }
 
+  void AbortSQL(){
```

Comment:
to be rather added to the GDALDataset object.

You could leave that one (in addition to the one of GDALDatset), and just call GDALDatasetAbortSQL(), since a OGRDatasetH is a GDALDatasetH

Reply:
I'm not sure I get it right, please have a look to my last commits if they address this issue. Also, I'm not sure about how to test the GDALDatset path, it seems to me that the python tests are using the Datasource object and not the dataset.

Reply:
> I'm not sure about how to test the GDALDatset path

you need to open the dataset with ds = gdal.OpenEx(filename, gdal.OF_VECTOR)

Reply:
... and add the test case to `ogr_pg.py`/`ogr_gpkg.py` or does it belong another test files? It seems to me that it should **not** be under `gdal/autotest/ogr`.

Reply:
>  It seems to me that it should **not** be under `gdal/autotest/ogr`.

in theory yes, but I'm not sure a lifetime would be enough to erase the OGR vs GDAL dichotomy :-) The tests are fine where you've added them.